apps/framework-cli/src/project.rs

```diff
@@ -173,16 +169,13 @@ impl Project {
     }
 
     pub fn load(directory: &PathBuf) -> Result<Project, ConfigError> {
-        let mut project_file = directory.clone();
-        project_file.push(PROJECT_CONFIG_FILE);
+        let project_file = directory.join(PROJECT_CONFIG_FILE);
 
         let mut project_config: Project = Config::builder()
-            // TODO: consider putting the defaults into a source (e.g. include_str a toml file)
-            .set_default(
-                "clickhouse_config.native_port",
-                ClickHouseConfig::default().native_port,
-            )
-            .unwrap() // key in set_default is a static string, this never fails
+            .add_source(File::from_str(
+                include_str!("default_config.toml"),
+                FileFormat::Toml,
+            ))
```

Comment:
I think this beats calling `.set_default().unwrap()` for every field we want to provide a default.
But then we provide the default values twice.

Reply:
Yeah I don't know which one is more prone to errors

Reply:
Do we actually need this all together? the default implementation should cover it - no?

Reply:
Unfortunately the config loader is not that smart.

Reply:
`Setting` is using the same config loader and we don't have to do this for default settings. example is the feature flags that are defaulted

Reply:
Ah I remember, it can use the default if the whole struct is absent. But if only one field is provided it crashes.

Reply:
Yeah you need to add the default pragma on the field also 

example:
https://github.com/514-labs/moose/blob/39324350287222a6887d9fffb912f4d6662dc3d3/apps/framework-cli/src/cli/settings.rs#L48-L49

Reply:
Yeah I was hoping that the it can merge the fields in the default struct value.

Reply:
With that in mind - can you remove `default_config.toml` then?